### PR TITLE
Configure target for clang thread safety checking

### PR DESCRIPTION
**Pull request by gaul**

### CODE REVIEW

**Comment by gaul** [comment]
This demonstrates how to configure static lock checking on Fedora which is what I use.  Instead we could create a new target for static lock checking.  What do you think?

**Comment by ggtakec** [comment]
@gaul I think it's a good idea to create a CI target just for static lock checking.
As mentioned in the previous topic, I'm thinking of separating out cppcheck/shellcheck as well.

File: .github/workflows/linux-ci-helper.sh
Comment (gaul) [reply .github/workflows/linux-ci-helper.sh]: Done.

**Comment by gaul** [comment]
Leaving as a draft until #2492 merges.

**Comment by gaul** [comment]
This is not passing flags correctly:

```
clang++ -DHAVE_CONFIG_H -I. -I..  -I/usr/include/fuse -D_FILE_OFFSET_BITS=64 -I/usr/include/libxml2 -DWITH_GZFILEOP  -DUSE_OPENSSL_30   -Wall -fno-exceptions -D_FILE_OFFSET_BITS=64 -D_FORTIFY_SOURCE=3 -std=c++11 -O -DS3FS_PTHREAD_ERRORCHECK=1 -MT cache.o -MD -MP -MF .deps/cache.Tpo -c -o cache.o cache.cpp
```

**Comment by gaul** [comment]
I correct multiple quoting issues and this now correctly finds locking issues.  It cannot merge until we fix the outstanding warnings.

**Comment by gaul** [comment]
Let's merge this before any other locking changes so it can catch any locking regressions.

**Review by ggtakec — APPROVED** [review]

**Pull request merged**
